rest_client.go

```diff
@@ -425,6 +434,27 @@ func (p *restProcess) RegisterSignalTrigger(_ context.Context, _ SignalTrigger)
 	return errors.New("cannot register signal trigger on remote processes")
 }
 
+func (p *restProcess) RegisterSignalTriggerID(ctx context.Context, triggerID SignalTriggerID) error {
+	req, err := http.NewRequest(http.MethodPatch, p.client.getURL("/process/%s/signal-trigger/%d", p.id, triggerID), nil)
+	if err != nil {
+		return errors.Wrap(err, "problem building request")
+	}
+
+	req = req.WithContext(ctx)
+
+	resp, err := p.client.client.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "problem making request")
+	}
+	defer resp.Body.Close()
```

Comment:
should we wrap up more of lines 439-449 into a function that we could reuse more places (like handle error?) also what does handleError do?

Reply:
I replaced this with `doRequest`. Since there are a lot of places that do essentially the same thing, I changed all those places to use `doRequest`.